### PR TITLE
Use substring instead of replaceFirst in OAuth2AuthorizationConsent

### PR DESCRIPTION
**Pull request by heartape**
Close https://github.com/spring-projects/spring-authorization-server/issues/1222

### CODE REVIEW

File: oauth2-authorization-server/src/main/java/org/springframework/security/oauth2/server/authorization/OAuth2AuthorizationConsent.java [hunk 1]
```diff
@@ -93,7 +93,7 @@ public Set<String> getScopes() {
 		Set<String> authorities = new HashSet<>();
 		for (GrantedAuthority authority : getAuthorities()) {
 			if (authority.getAuthority().startsWith(AUTHORITIES_SCOPE_PREFIX)) {
-				authorities.add(authority.getAuthority().replaceFirst(AUTHORITIES_SCOPE_PREFIX, ""));
+				authorities.add(authority.getAuthority().substring(AUTHORITIES_SCOPE_PREFIX.length()));
```
Comment (jgrandja) [inline oauth2-authorization-server/src/main/java/org/springframework/security/oauth2/server/authorization/OAuth2AuthorizationConsent.java:96]: > I conducted a simple test and achieved a 40 times performance improvement.

How did you conduct the test?

**Review by jgrandja — CHANGES_REQUESTED** [review]
Thanks for the PR @heartape. Please see review comments.

File: oauth2-authorization-server/src/main/java/org/springframework/security/oauth2/server/authorization/OAuth2AuthorizationConsent.java (hunk 1 above)
Comment (heartape) [reply oauth2-authorization-server/src/main/java/org/springframework/security/oauth2/server/authorization/OAuth2AuthorizationConsent.java:96]: This is all the testing information：

```java
public static void main(String[] args) {
    String scope = "SCOPE_";
    String email = "SCOPE_email";
    int count = 10000000;

    long start1 = System.currentTimeMillis();
    for (int i = 0; i < count; i++) {
        String s = email.replaceFirst(scope, "");
    }
    long end1 = System.currentTimeMillis();
    System.out.println("replaceFirst : " + (end1 - start1));

    long start2 = System.currentTimeMillis();
    for (int i = 0; i < count; i++) {
        String s = email.substring(scope.length());
    }
    long end2 = System.currentTimeMillis();
    System.out.println("substring : " + (end2 - start2));
    System.out.println("count : " + count);
}
```
```
windows(8c16t 2G) idea 
java version "17.0.3.1" 2022-04-22 LTS
Java(TM) SE Runtime Environment (build 17.0.3.1+2-LTS-6)
Java HotSpot(TM) 64-Bit Server VM (build 17.0.3.1+2-LTS-6, mixed mode, sharing)

result:
replaceFirst : 2383
substring : 69
count : 10000000

###############################################################

centos7 vm(2c 2G)  command line 
java version "17.0.6" 2023-01-17 LTS
Java(TM) SE Runtime Environment (build 17.0.6+9-LTS-190)
Java HotSpot(TM) 64-Bit Server VM (build 17.0.6+9-LTS-190, mixed mode, sharing)

result:
replaceFirst : 2856
substring : 107
count : 10000000
```

File: CONTRIBUTING.adoc [hunk 2]
```diff
@@ -67,7 +67,7 @@ Use git rebase –interactive, git add –patch and other tools to "squash" mult
 
 == Format commit messages
 
-. Keep the subject line to 50 characters or less if possible.
+. Keep the subject line to 50 characters or fewer if possible.
```
[the remark this answers is not in the document]
Comment (heartape) [reply CONTRIBUTING.adoc]: ok

File: oauth2-authorization-server/src/main/java/org/springframework/security/oauth2/server/authorization/oidc/web/authentication/OidcLogoutAuthenticationConverter.java [hunk 3]
```diff
@@ -65,12 +65,6 @@ public Authentication convert(HttpServletRequest request) {
 			principal = ANONYMOUS_AUTHENTICATION;
 		}
 
-		String sessionId = null;
```
[the remark this answers is not in the document]
Comment (heartape) [reply oauth2-authorization-server/src/main/java/org/springframework/security/oauth2/server/authorization/oidc/web/authentication/OidcLogoutAuthenticationConverter.java]: ok

File: CONTRIBUTING.adoc (hunk 2 above)
[the remark this answers is not in the document]
Comment (heartape) [reply CONTRIBUTING.adoc]: @jgrandja I will submit a new PR for this section. Do you think this is a good idea?

[the remark this answers is not in the document]
Comment (jgrandja) [reply CONTRIBUTING.adoc]: The use of "less" is proper grammar so we will leave as-is

**Comment by jgrandja** [comment]
Thanks for the update @heartape. This is now merged.

**Pull request closed without merging**